superset-frontend/src/dashboard/containers/DashboardPage.tsx

```diff
@@ -192,7 +239,8 @@ export const DashboardPage: FC<PageProps> = ({ idOrSlug }: PageProps) => {
   }, [addDangerToast, datasets, datasetsApiError, dispatch]);
 
   if (error) throw error; // caught in error boundary
-  if (!readyToRender || !hasDashboardInfoInitiated) return <Loading />;
+
+  const isLoading = !dashboardInfo;
```

Comment:
Must decouple the data that is required to show the charts list from the data required to show other parts of the Dashboard to enable progressive rendering.